src/Microsoft.ApplicationInsights.AspNetCore/Logging/Implementation/ApplicationInsightsLogger.cs

```diff
@@ -41,19 +45,21 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
         {
             if (this.IsEnabled(logLevel))
             {
-                var dict = new Dictionary<string, string>();
+                TraceTelemetry traceTelemetry = new TraceTelemetry(formatter(state, exception), this.GetSeverityLevel(logLevel));
+                IDictionary<string, string> dict = traceTelemetry.Context.Properties;
                 dict["CategoryName"] = this.categoryName;
                 dict["Exception"] = exception?.ToString();
-                var stateDictionary = state as IReadOnlyList<KeyValuePair<string, object>>;
+                IReadOnlyList<KeyValuePair<string, object>> stateDictionary = state as IReadOnlyList<KeyValuePair<string, object>>;
                 if (stateDictionary != null)
                 {
-                    foreach (var item in stateDictionary)
+                    foreach (KeyValuePair<string, object> item in stateDictionary)
                     {
                         dict[item.Key] = Convert.ToString(item.Value);
                     }
                 }
 
-                this.telemetryClient.TrackTrace(formatter(state, exception), this.GetSeverityLevel(logLevel), dict);
+                traceTelemetry.Context.GetInternalContext().SdkVersion = this.sdkVersion;
```

Comment:
Then I would prefer having this `traceTelemetry.Context.GetInternalContext().SdkVersion = ` and ` this.sdkVersion = SdkVersionUtils.VersionPrefix + SdkVersionUtils.GetAssemblyVersion();` in common utility class. Just don't like increasing code duplication